clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java

```diff
@@ -71,12 +71,12 @@
     /**
      * @see KafkaConsumer#commit(Map, CommitType)
      */
-    public void commit(Map<TopicPartition, Long> offsets, CommitType commitType);
+    public void commit(Map<TopicPartition, OffsetMetadata> offsets, CommitType commitType);
```

Comment:
I was thinking that the user would maintain `Map<TopicPartition, OffsetMetadata>`, so map.values() would return `Collection<OffsetMetadata>`, which could be passed directly to commit. That said, I don't have a strong preference on any of the choices so far, but I'm probably leaning toward the initial approach. The overhead is not so high and users who want to manage their own commits may find metadata more useful than we're imagining. Off the top of my head, I can imagine wanting to add timestamps, hostnames, or maybe versioning information to the commit. If instead it's really such an extreme edge case that almost no users would find value in, maybe it doesn't belong in the API at all and we can think about another way to expose it.